burotel/indico_burotel/plugin.py

```diff
@@ -88,3 +152,29 @@ def _inject_time(self, sender, data, **kwargs):
             _add_missing_time('start_dt', data, time(0, 0))
         if 'end_dt' in data:
             _add_missing_time('end_dt', data, time(23, 59))
+
+    def _booking_created(self, booking, **kwargs):
+        if booking.state == ReservationState.accepted:
+            _check_no_parallel_bookings(booking)
+            _check_update_permissions(booking)
+
+    def _booking_modified(self, booking, changes, **kwargs):
+        if (
+            booking.state != ReservationState.accepted or
+            booking.room.get_attribute_value('electronic-lock') != 'yes'
+        ):
+            return
+
+        if 'end_dt/date' in changes or 'start_dt/date' in changes:
+            @after_this_request
```

Comment:
Just wondering, will this callback also trigger when there's an error later during the request?

Reply:
Indeed. Not sure how to keep that from happening, though ☹️

Reply:
It might not matter that much in the end as the changes won't be committed when this happens, and since we load the booking fresh from the database in celery, it would not cause any inconsistency.

Reply:
Yeah, but ADAMS is still contacted, right?

Reply:
if the booking creation wasn't committed, the task will fail because it cannot query the booking from the DB, and if a modification wasn't committed it'll just update it with the same data as it had before

Reply:
Ah true... there will be a rollback, so it's fine. Thanks.